frontend/src/components/Buttons/TypeButton.vue

```diff
@@ -1,94 +1,84 @@
-        v-bind="attrs"
-        v-on="on">
-        {{ items[model].name }}
+        v-bind="props">
+        {{
+          model == 0
+            ? items[0].title
+            : items.find((x) => x.value == model[0]).title
+        }}
         <v-icon end>
           <i-mdi-menu-down />
         </v-icon>
       </v-btn>
-      <v-btn
-        v-else
-        :disabled="disabled"
-        class="my-2"
-        icon
-        v-bind="attrs"
-        v-on="on">
+      <v-btn v-else :disabled="disabled" class="my-2" icon v-bind="props">
         <v-icon>
           <i-mdi-eye />
         </v-icon>
       </v-btn>
     </template>
-    <v-list dense>
-      <v-list-group
-        v-model="model"
-        @change="$emit('change', items[model].value)">
-        <v-list-item v-for="item in items" :key="item.value">
-          <v-list-item-title>{{ item.name }}</v-list-item-title>
-        </v-list-item>
-      </v-list-group>
-    </v-list>
+    <v-list
+      v-model:selected="model"
+      :items="items"
+      @update:selected="emit('change', model[0])" />
   </v-menu>
 </template>
 
-<script lang="ts">
-import { defineComponent } from 'vue';
+<script setup lang="ts">
+import { defineProps, computed, ref } from 'vue';
+import { useI18n } from 'vue-i18n';
 
-export default defineComponent({
-  props: {
-    type: {
-      type: String,
-      required: true
-    },
-    disabled: {
-      type: Boolean,
-      required: false,
-      default: false
-    }
-  },
-  data() {
-    return {
-      model: 0
-    };
+const { t } = useI18n();
+const props = defineProps({
+  type: {
+    type: String,
+    required: true
   },
-  computed: {
-    items(): Array<Record<string, string>> {
-      switch (this.type) {
-        case 'movies': {
-          return [
-            { name: this.$t('movies'), value: 'Movie' },
-            { name: this.$t('collections'), value: 'BoxSet' },
-            { name: this.$t('actors'), value: 'Actor' },
-            { name: this.$t('genres'), value: 'Genre' },
-            { name: this.$t('studios'), value: 'Studio' }
-          ];
-        }
-        case 'music': {
-          return [
-            { name: this.$t('albums'), value: 'MusicAlbum' },
-            { name: this.$t('artists'), value: 'MusicArtist' },
-            { name: this.$t('genres'), value: 'MusicGenre' }
-          ];
-        }
-        case 'tvshows': {
-          return [
-            { name: this.$t('series'), value: 'Series' },
-            { name: this.$t('actors'), value: 'Actor' },
-            { name: this.$t('genres'), value: 'Genre' },
-            { name: this.$t('networks'), value: 'Studio' }
-          ];
-        }
-        default: {
-          return [];
-        }
-      }
+  disabled: {
+    type: Boolean,
+    required: false,
+    default: false
+  }
+});
+const emit = defineEmits<{
+  (e: 'change', types: Record<string, string[]>): void;
+}>();
+var model = ref(0);
+
+const items = computed((): Array<Record<string, string>> => {
```

Comment:
Type inference is automatic, so this is redundant. However, in case TS can't infer the types properly, it's this way:
```suggestion
const items = computed<Array<Record<string, string>>>(()  => {
```